### PR TITLE
Add new definitions for upyun package

### PR DESCRIPTION
**Pull request by Groupguanfang**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/upyun/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,331 @@
+// Type definitions for upyun 3.4
+// Project: https://github.com/upyun/node-sdk/
+// Definitions by: Zero <https://github.com/groupguanfang>, Leigh <https://github.com/upyun>
```
Comment (peterblazejewicz) [inline types/upyun/index.d.ts:3]:
```suggestion
// Definitions by: Zero <https://github.com/groupguanfang>
//                 Leigh <https://github.com/upyun>
```
just for clarity

File: types/upyun/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+/// <reference types="node" />
```
Comment (peterblazejewicz) [inline types/upyun/index.d.ts:5]:
```suggestion

/// <reference types="node" />
```

File: types/upyun/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+
+import buffer = require('buffer');
```
Comment (peterblazejewicz) [inline types/upyun/index.d.ts:7]: is this `node:buffer` (and can you use scoped imports?, there is a `buffer` package)

**Review by peterblazejewicz — APPROVED** [review]
LGMT!
@Groupguanfang thanks!

**Comment by Groupguanfang** [comment]
Ready to merge

**Pull request merged**